saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -94,14 +94,21 @@ def perform_mutation(cls, _root, info, ids, **data):
                 variant__in=variants, order__status=OrderStatus.DRAFT
             ).values_list("pk", flat=True)
         )
-
+        data["manager"] = info.context.plugins
         response = super().perform_mutation(_root, info, ids, **data)
 
         # delete order lines for deleted variants
         order_models.OrderLine.objects.filter(pk__in=order_line_pks).delete()
 
         return response
 
+    @classmethod
+    def bulk_action(cls, queryset, manager):
+        products = [product for product in queryset]
+        queryset.delete()
+        for product in products:
+            manager.product_deleted(product)
```

Comment:
AFAIK The product here losts relations to own variants. I would say that providing at least an ID or slug of variants is really important to proper sync between systems.
```
products = [product for product in Product.objects.filter(name__icontains="Juice")]
In [60]: products[0].variants.all()
Out[60]: <ProductVariantQueryset [<ProductVariant: 1l>, <ProductVariant: 2l>, <ProductVariant: 500ml>]>
Number of queries 1
In [61]: Product.objects.filter(name__icontains="Juice").delete()
In [62]: products[0].variants.all()
Out[62]: <ProductVariantQueryset []>
Number of queries 1
```